### DISCUSSION TITLE
rpm 4.19 multi x86-64 arch versions

### DISCUSSION THREAD

**Discussion by finjulhich** [rpm-software-management/rpm#2825] [Q&A]
I have a few hosts with cpus of different x86-64 levels 2 3 and 4.
I am working on producing rpms via cmake.

1. Can I produce **fat** rpm files, ie at install time, the correct version (arch level) of the binaries get installed?
2. Do provides/require take into account these levels?
3. Is this concept expandable to entirely other archs like ARM?
Rds,

**Comment by pmatilai** [comment]
The arch levels are not a new feature, just new sub-architectures. Think i386 getting expanded to i486, i586 and i686 back then, and similar levels exist on arm already. The rpm arch only works as package level differentiator and to ensure you can't install a package eg for level 4 on a level 3 host.

**Reply by finjulhich** [reply]
Ok so in order to install on cpus of different levels, I need to produce rpms of differnet levels.
The level is encoded in the rpm file name?

**Reply by pmatilai** [reply]
Again, these are just (sub)architectures, and as always, the arch is present in the rpm package file name.
